### ISSUE TITLE
Scanning private git repositories

### ISSUE THREAD

**Issue by truedat101** [zricethezav/gitleaks#30]
Would be very useful to be able to provide http credentials as an option to authenticate with private git repositories.  Currently does not appear to support anything but public projects, unless I have misunderstood the options.

**Comment by zricethezav** [comment]
Agreed this would be ideal to have. Private repo scanning is supported via ssh keys

**Comment by zricethezav** [comment]
Now that I think about it, probably easiest to revert go-git changes and just shell the clone

**Comment by truedat101** [comment]
bingo.  Yes.  My local git already works automagically using git credential manager for windows (https://github.com/Microsoft/Git-Credential-Manager-for-Windows) if I am using https.  For ssh that would be the same under a linux setup.

**Comment by zricethezav** [comment]
reverted https://github.com/zricethezav/gitleaks/pull/31

**Comment by truedat101** [comment]
Hmmm, not exactly.  It does the following on windows:

- Triggers the Git Login Window (I enter my credentials)
- Then triggers an open ssh login window (I enter my credentials)
- fails

I think it may be this is difficult to support on windows with the current design if it assumes ssh login and the user has not set up ssh keys on windows.

**Comment by zricethezav** [comment]
Hopefully the version bump will help this @truedat101 
https://github.com/zricethezav/gitleaks/releases/tag/v0.2.0
Havent tested on windows... lemme know

**Comment by lanmalkieri** [comment]
This appears to still trigger a request for a login. 

Tried running with both GIT_TOKEN env var set and manually specifying token at the cli --token argument, prompts me with github username.

The following is set with env var set:
stobiewan|[]:[~/go] > env|grep GITHUB_TOKEN
GITHUB_TOKEN=REDACTED

REDACTED|[]:[~/go] > bin/gitleaks --json https://github.com/REDACTED/onboarding
Cloning https://github.com/REDACTED/onboarding...
Username for 'https://github.com':

**Comment by truedat101** [comment]
I haven't tried the new update, however, the net effect of the previous version I tried was that I had to re-run Windows GIT credential manager to get git working again with github.  I will try this again later in the week with the new version.

**Comment by supanadit** [comment]
https://github.com/src-d/go-git/issues/1228#issue-508250464
